From b4b982d19846859cb7f70343abf382246c54100a Mon Sep 17 00:00:00 2001
From: "Song, Yoong Siang" <yoong.siang.song@intel.com>
Date: Thu, 6 Aug 2020 00:08:02 +0800
Subject: [PATCH 1/7] net: stmmac: Fix runtime PM build error

Fix struct dev_pm_info's member not found build error when
CONFIG_PM is not defined.

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Song, Yoong Siang <yoong.siang.song@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 95c6795..d6c43b8 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -6924,6 +6924,7 @@ int stmmac_dvr_probe(struct device *device,
 		return ret;
 #endif
 
+#ifdef CONFIG_PM
 	/* To support runtime PM, we need to make sure usage_count is equal to 0
 	 * when runtime_auto flag is set. Otherwise, it should be equal to 1.
 	 */
@@ -6934,6 +6935,7 @@ int stmmac_dvr_probe(struct device *device,
 		while (atomic_read(&priv->device->power.usage_count) > 1)
 			pm_runtime_put_noidle(device);
 	}
+#endif
 
 	return ret;
 
-- 
2.7.4

